controlplane/pkg/model/client_connection.go

```diff
@@ -134,20 +138,19 @@ func (d *clientConnectionDomain) GetAllClientConnections() []*ClientConnection {
 	return rv
 }
 
-func (d *clientConnectionDomain) DeleteClientConnection(id string) {
-	d.delete(id)
-}
-
-func (d *clientConnectionDomain) UpdateClientConnection(cc *ClientConnection) {
-	d.store(cc.ConnectionID, cc)
+func (d *clientConnectionDomain) DeleteClientConnection(id string) error {
+	if ok := d.delete(id); ok {
+		return nil
+	}
+	return fmt.Errorf("trying to delete client connection by not existing id: %v", id)
 }
 
-func (d *clientConnectionDomain) ApplyClientConnectionChanges(id string, f func(*ClientConnection)) *ClientConnection {
```

Comment:
Both variants are actually not so good, we need to track after call if anyone changed our model value.

Reply:
Probable compare and swap should be applied.

Reply:
I guess we still need `ApplyClientConnectionChanges` for some rare cases when we only need to update some parameter without any knowledge about what do we have in model.
Also I have added `CompareAndSwapClientConnection` : 88ced90

Reply:
The approach was ultimately reworked with `model.ClientConnectionEditor`.